File: wave.m
%----------小波变换--------
%-----分解1层---------
%----信号data--
[cA,cD]=dwt(data,'haar');   %cA:尺度系数，cD细节系数
%重构
newdata=idwt(cA,cD,'haar');%重构
newdata=idwt(cA,[],'haar');%获得尺度系数的信号
newdata=idwt([],cD,'haar');%获得细节系数的信号
%--------分解多层-----
[c,l]=wavedec(data,3,'haar');
newdata=waverec(c,l,'haar');%重构
newdata=wrcoef('a',c,l,3);%计算第3层的尺度系数的信号
newdata=wrcoef('d',c,l,3);%计算第3层的细节系数的信号

File: wave.py
import pywt
#--信号data---
cA,cD1,cD2,cD3=pywt.wavedec(data,'haar',level=3)  #分解
data=pywt.waverec([cA,cD1,cD2,cD3],'haar')    #重构
x1=pywt.upcoef('a',cA,'haar',level=3,take=N)   #计算第3层尺度系数的信号，长度为N
x2=pywt.upcoef('d',cD1,'haar',level=3,take=N)   #计算第3层细节系数的信号，长度为N
